dwave/cloud/computation.py

```diff
@@ -487,6 +491,43 @@ def cancel(self):
                 self._cancel_sent = True
                 self.solver.client._cancel(self.id, self)
 
+    def get_id(self, timeout=None):
+        """Return the submitted problem ID. Block until the ID becomes known,
+        or until `timeout` expires.
+
+        Args:
+            timeout (float, default=None):
+                Timeout in seconds. By default, wait indefinitely for problem
+                id to become known/available.
+
+        Returns:
+            str:
+                Problem ID, as returned by SAPI.
+
+        Raises:
+            :exc:`concurrent.futures.TimeoutError`:
+                When `timeout` exceeded, and problem id not ready.
+
+        """
+        if not self._id_ready_event.wait(timeout=timeout):
+            raise TimeoutError("problem id not available yet")
+
+        return self.id
+
+    def set_id(self, id_):
```

Comment:
You could consider doing this in a setter rather than a method. That was there would still be one way to set, but two ways to get (legacy and blocking).

Reply:
@arcondello, I like that idea. I just don't like the asymmetry that it brings: `id` setter would be a pair with `get_id` (they both use the `Event`), and then there would be the plain `id` getter...

Reply:
Maybe renaming `get_id` to something like `wait_for_id` would help there.

Reply:
I like that, makes it more explicit